main.py

```diff
@@ -60,13 +154,92 @@ async def check_scores_task(self):
+            diff_str = f" [x{mult}]"
+
+        player = score["user_name"]
+        run_length = round(score["value"], 3)
+        rank = score["position"]
+
+        if pack_name[0] == "#":
+            pack_name = "\\" + pack_name
+
+        return f"**{pack_name} - {level_name}{diff_str}** <:hexagon:1432891341297418322> **{player}** achieved **#{rank}** with a score of **{run_length}**"
+
+    async def send_recent_score(self, scores_json, saved_state, player_check):
+        # because we are just sending one score, we don't have to check for edits or update the saved state
+        channels = self.get_output_channels(saved_state["subscribed_channels"])
+        for i in range(len(scores_json)):
+            score = scores_json[len(scores_json) - i - 1]
+            player = score["user_name"]
+
+            if player == player_check or player_check == None:
+                score_text = self.get_score_text(score)
+                return score_text
+        return "No recent scores found. (within the last six hours)"
+
+    async def send_best_score(self, saved_state, pack, level, difficulty_mult):
+        channels = self.get_output_channels(saved_state["subscribed_channels"])
+
+        # the order of operations for this is that we need to lookup the pack id, then find the best score, then run get_score_text
+
+        for pack_id, pack_entry in self.pack_lookup.items():
+            if pack_entry["pack_name"].lower() == pack.lower():
+                for level_id, level_entry in pack_entry["levels"].items():
+                    if level_entry[0].lower() == level.lower():
+                        pack_ID = pack_id
+                        level_ID = level_id
+
+        # this is the "finding the best score" part
+        pack_ID_str = urllib.parse.quote(pack_ID)
+        level_ID_str = urllib.parse.quote(level_ID)
+        # bad way of doing this!
+        level_options_str = urllib.parse.quote(f"{{difficulty_mult: {difficulty_mult}}}")
+
+        try:
+            lb_scores = requests.get(f"{LB_API_SERVER}/get_leaderboard/{pack_ID_str}/{level_ID_str}/{level_options_str}").json()
+        except:
+            log(f"WARNING: Could not get leaderboard for {LB_API_SERVER}/get_leaderboard/{pack_ID_str}/{level_ID_str}/{level_options_str}.")
+            return "Unable to fetch leaderboard. (did you type in the level correctly?)"
+
+        for score in lb_scores:
+            if score["position"] == 1:
+                # running this function here is quite redundant
+                score_text = self.get_score_text(score)
+                return score_text
+        return "This level has no scores."
+
     async def send_wrs(self, scores_json, saved_state):
-        channel = self.get_output_channel()
+        # note: the channels are only checked at the start of the loop, meaning that scores could be sent even after running the unsubscribe command
+        channels = self.get_output_channels(saved_state["subscribed_channels"])
```

Comment:
I realize that this is a draft, but for the sake of completeness (of the review) I will point out that this should be implemented.